plugins/thunderbolt/fu-plugin-thunderbolt.c

```diff
@@ -193,23 +193,29 @@ fu_plugin_thunderbolt_udev_get_version (GUdevDevice *udevice)
 static gboolean
 fu_plugin_thunderbolt_is_native (GUdevDevice *udevice, gboolean *is_native, GError **error)
 {
+	gsize nr_chunks;
 	g_autoptr(GFile) nvmem = NULL;
 	g_autoptr(GBytes) controller_fw = NULL;
-	gchar *content;
-	gsize length;
+	g_autoptr(GInputStream) istr = NULL;
 
 	nvmem = fu_plugin_thunderbolt_find_nvmem (udevice, TRUE, error);
 	if (nvmem == NULL)
 		return FALSE;
 
-	if (!g_file_load_contents (nvmem, NULL, &content, &length, NULL, error))
+	/* read just enough bytes to read the status byte */
+	nr_chunks = (FU_TBT_OFFSET_NATIVE / FU_TBT_CHUNK_SZ) + 1;
```

Comment:
In this case it works, but if the offset would be changed to a multiple of CHUNK_SZ this reads 1 extra chunk over what is needed. Your call.

Reply:
Ya, i did think of maybe doing a `nr_chunks++` if the offset was not `% FU_TBT_CHUNK_SZ` but decided it wasn't really worth it. If you know of a clever bitmask shifting thing that can round up to 64 without lots of divide/multiply, I'm all ears :)

Reply:
As long as everything is const, it's done on compile time anyway.
But why not just using what I suggested originally?
`nr_chunks = ((FU_TBT_OFFSET_NATIVE + FU_TBT_CHUNK_SZ - 1) / FU_TBT_CHUNK_SZ);`

Reply:
It would appear the compiler is cleverer than I am. I'll use your suggestion, thanks!